chainer/_backprop_utils.py

```diff
@@ -53,7 +53,7 @@ def get_as_list(self, node):
             if self._load_if_new and node.creator_node is None:
                 node._check_old_style_gradient()
                 # accumulate the gradient only if the node is a leaf
-                grads[node] = _pure(node.grad_var)
+                grads[node] = _pure(node._pop_grad_var_if_available())
```

Comment:
`GradTable` is no longer a simple data holder but it describes the algorithm, because it modifies the external data structure.
This behavior is not clearly expressed and therefore difficult to understand the code, IMO.
Perhaps the method name `get_as_list` should be something that expresses that behavior.
Or perhaps the class name `GradTable` should be something that does not imply a simple data holder.

Reply:
- For the line, I'll move the logic from `get_as_list` to `_backprop_to_all`.
- `GradTable.pop` is not simple because of an issue that #5722 will fix.